cg/services/order_validation_service/workflows/tomte/validation/inter_field/rules.py

```diff
@@ -81,3 +83,10 @@ def validate_mothers_in_same_case_as_children(order: TomteOrder) -> list[MotherN
         case_errors = get_mother_case_errors(case)
         errors.extend(case_errors)
     return errors
+
+
+def validate_subject_ids_different_from_case_names(order: TomteOrder) -> list[CaseSampleError]:
+    errors = []
+    for case in order.cases:
+        errors.extend(validate_subject_ids_in_case(case))
```

Comment:
In general I think it is preferable to avoid nesting function calls like this

Reply:
Fixed